src/core/game.js

```diff
@@ -0,0 +1,184 @@
+const SpaceShip = require("../core/spaceShip");
+const Point = require("../utils/point");
+const mathSupport = require("../utils/mathSupport");
+const Galaxy = require("./galaxy");
+
+class Game {
+    constructor(config) {
+        this.gameConfig = config;
+        this.player = new SpaceShip(new Point(3, 3), new Point(3, 3), config.MAX_POWER);
+        this.galaxy = new Galaxy(config.GALAXY);
+        this.isOver = false;
+        this.starDates = config.INITIAL_STARDATES;
+        this.knownGalaxy = [];
+        for (let i = 0; i < Math.sqrt(config.GALAXY.QUADRANTS); i++) {
+            this.knownGalaxy.push([]);
+            for (let j = 0; j < Math.sqrt(config.GALAXY.QUADRANTS); j++) {
+                this.knownGalaxy[i].push(["?", "?", "?"]);
+            }
+        }
+    }
+
+    makeKlingonsTurn() {
+        this.galaxy.removeDestroyed();
+        this.galaxy.getQuadrantObjects(this.player.quadrant).forEach((so, i) => {
+            if (so.type == "ship" ) {
+                let distanceToPlayer = mathSupport.cityBlockDistance(
+                    this.player.sector.x,
+                    this.player.sector.y,
+                    so.sector.x,
+                    so.sector.y
+                );
+                let dmg = 30;
+                so.reducePower(dmg);
+                let hasHit = Math.random() < (5 / (distanceToPlayer + 4));
+                if(hasHit){
+                    console.log(`Klingons has attacked and hit you! You lose ${dmg} power.`);
+                    this.player.reducePower(dmg);
+                } else {
+                    console.log('Klingons has attacked, but missed.');
+                }
+            }
+        });
+    }
+
+    playerShoot(power){
+        if(power >= this.player.power){
```

Comment:
Self destruction with `power > this.player.power` would be interesting option for a tie result.